virttest/remote_commander/remote_master.py

```diff
@@ -259,6 +265,15 @@ def listen_errors(self, cmd):
                             remote_interface.MessengerError)):
             raise cmd
 
+    def listen_queries(self, cmd):
+        """Manage queries from slave side.
+        """
```

Comment:
OK so this basically adds another type of communication called "queries", which the user can use inside the the remoter to ask the master for response and you are using it to notify your avocado-vt test about something from the test inside the VM.

I have to say that I'd find running `aexpect` session and using `print` and `raw_input` as simpler and easier to debug solution, but if this fits you, why not.